Shopify/vscode-ruby-lsp
# Add gem actions in the Gemfile

### Motivation

See: https://github.com/Shopify/ruby-lsp/issues/585
Link to: https://github.com/Shopify/ruby-lsp/pull/710

### Implementation

- Register the new command `rubyLsp.openGem`
- When the command is triggered open the link through `openExternal`

I had to filter lenses before calling `createTestItems` because it can crash now that we receive two different kinds of codelens which are not formatted in the same way. For this reason, this pull request should be merged **before** server one.

### Automated Tests

I don't know how to test it. But feel free to point me in the right direction 🙏 

### Manual Tests

- Check out this branch
- Checkout ruby-lsp branch 
- Open a Gemfile
- Something like that is expected:

<details><summary>Record</summary>

![Screen-Recording-2023-05-09-at-22 37 45](https://github.com/Shopify/vscode-ruby-lsp/assets/38727166/71898ac7-e6f3-42f4-91ff-56f016dab175)
</details>
